src/formatDate.js

```diff
@@ -8,7 +8,66 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYear = yearFormatAndPlace(fromFormat);
+  const newYear = yearFormatAndPlace(toFormat);
+
+  const formatedDate = [];
+  let year = '';
+
+  const dateParts = date.split(fromFormat[3]);
+  const newDateSeparator = toFormat[3];
+
+  const thresholdYear = 30;
+
+  if (oldYear.format !== newYear.format) {
```

Comment:
There is no check to ensure that `fromFormat` and `toFormat` are of the correct length. It's assumed they both contain 4 elements including the separator, but if they don't, this could lead to runtime errors.